pallets/aura-ext/src/consensus_hook.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The definition of a [`ConsensusHook`] trait for consensus logic to manage
+//! block velocity.
+//!
+//! The velocity `V` refers to the rate of block processing by the relay chain.
+
+use sp_std::num::NonZeroU32;
+
+/// Block processing velocity, essentially defining the limit of block production per
+/// aura slot.
+#[derive(Debug, Copy, Clone)]
+pub struct BlockProcessingVelocity(NonZeroU32);
+
+impl BlockProcessingVelocity {
+	pub(crate) fn get(&self) -> u32 {
+		self.0.get()
+	}
+}
+
+impl From<NonZeroU32> for BlockProcessingVelocity {
+	fn from(value: NonZeroU32) -> Self {
+		Self(value)
+	}
+}
+
+/// The consensus hook for managing number of blocks authored per slot.
+pub trait ConsensusHook {
+	/// This hook is called on each block initialization to determine if it's valid to author
+	/// another block.
+	fn block_processing_velocity() -> BlockProcessingVelocity;
```

Comment:
Is there any argument we could expect? Looks like the only option is some constant value, and if para takes another core they'd have to upgrade their runtime

Reply:
Yeah. It'll have to be constant for now. This'll be a relatively easy amendment once we have completed the design of elastic scaling.